sql/parser/builtins.go

```diff
@@ -347,6 +348,80 @@ var builtins = map[string][]builtin{
 	"max": aggregateImpls(boolType, intType, floatType, stringType),
 	"min": aggregateImpls(boolType, intType, floatType, stringType),
 	"sum": aggregateImpls(intType, floatType),
+
+	// Timestamp/Date functions.
+
+	"age": {
+		builtin{
+			types:      typeList{timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: time.Now().Sub(args[0].(DTimestamp).Time)}, nil
+			},
+		},
+		builtin{
+			types:      typeList{timestampType, timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: args[0].(DTimestamp).Sub(args[1].(DTimestamp).Time)}, nil
+			},
+		},
+	},
+
+	"current_date": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyDate,
+			fn: func(args DTuple) (Datum, error) {
+				return DDate{Time: time.Now().Truncate(24 * time.Hour)}, nil
+			},
+		},
+	},
+
+	"clock_timestamp":   {nowBuiltin()},
+	"current_timestamp": {nowBuiltin()},
+	"localtimestamp":    {nowBuiltin()},
+	"now":               {nowBuiltin()},
+
+	"extract": {
+		builtin{
+			types:      typeList{stringType, timestampType},
+			returnType: DummyFloat,
+			fn: func(args DTuple) (Datum, error) {
+				t := args[1].(DTimestamp)
+				s := strings.ToLower(string(args[0].(DString)))
+				switch s {
+				case "year":
+					return DFloat(t.Year()), nil
+
+				case "month":
+					return DFloat(t.Month()), nil
+
+				case "day":
+					return DFloat(t.Day()), nil
+
+				case "hour":
+					return DFloat(t.Hour()), nil
+
+				case "minute":
+					return DFloat(t.Minute()), nil
+
+				case "second":
+					return DFloat(float64(t.Second()) + (float64(t.Nanosecond()) / 1000000000)), nil
+
+				case "millisecond":
+					return DFloat(float64(t.Second()*1000) + (float64(t.Nanosecond()) / 1000000)), nil
+
+				case "microsecond":
+					return DFloat(float64(t.Second()*1000*1000) + (float64(t.Nanosecond()) / float64(1000))), nil
+
+				default:
```

Comment:
What about `dayofweek`, `dayofyear`, `quarter`? If not easily implementable in this PR add a TODO.